gax-java/gax-httpjson/src/main/java/com/google/api/gax/httpjson/ProtoMessageRequestFormatter.java

```diff
@@ -50,35 +50,36 @@
   private final String rawPath;
   private final PathTemplate pathTemplate;
   private final FieldsExtractor<RequestT, Map<String, String>> pathVarsExtractor;
-  private final List<String> additionalRawPaths;
-  private final List<PathTemplate> additionalPathTemplates;
+  private final Map<PathTemplate, FieldsExtractor<RequestT, Map<String, String>>>
+      additionalPathsExtractorMap;
 
   private ProtoMessageRequestFormatter(
       FieldsExtractor<RequestT, String> requestBodyExtractor,
       FieldsExtractor<RequestT, Map<String, List<String>>> queryParamsExtractor,
       String rawPath,
       PathTemplate pathTemplate,
       FieldsExtractor<RequestT, Map<String, String>> pathVarsExtractor,
-      List<String> additionalRawPaths,
-      List<PathTemplate> additionalPathTemplates) {
+      Map<PathTemplate, FieldsExtractor<RequestT, Map<String, String>>>
+          additionalPathsExtractorMap) {
```

Comment:
This is a private constructor which should give us some flexibility to update the parameters.